### DISCUSSION TITLE
Rule names convention

### DISCUSSION THREAD

**Discussion by Shatur** [pest-parser/pest#948] [General]
I noticed that pest use snake_case for all its examples and documentation. But it results in having `Rule` with snake case members in Rust code. Why snake_case is used instead of CamelCase to make the generated enum look idiomatic in Rust?

**Comment by NyalephTheCat** [comment]
It's a good question... My best guess is that it's just because Pest was inspired on some other grammar that use snake case and rust uses camel case...

**Reply by Shatur** [reply]
Got it. I will probably use CamelCase in my code then.
